src/renderer/components/commands-action-button.tsx

```diff
@@ -175,9 +175,19 @@ export class GistActionButton extends React.Component<
     appState.activeGistAction = GistActionState.updating;
 
     try {
+      const {
+        data: { files: oldFiles },
+      } = await octo.gists.get({ gist_id: appState.gistId! });
+
+      const files = this.gistFilesList(values);
+      for (const id of Object.keys(oldFiles)) {
+        // Gist files are deleted by setting content to an empty string.
+        if (!Object.keys(files).includes(id)) files[id] = { content: '' };
```

Comment:
This seems inefficient -- one, we're rebuilding the array each time we iterate the loop, and two, we trade away any v8 optimizations for object property lookups and use an array linear search instead

```suggestion
        if (!(id in files)) files[id] = { content: '' };
```

Reply:
TIL the `in` operator exists 🙈